### ISSUE TITLE
How to set parameters un-trainable when using Keras as a simplified interface to TensorFlow?

### ISSUE THREAD

**Issue by HC-2016** [fchollet/keras#7621]
When using keras as a simplified interface in tensorflow, I set trainable to be false by
```
conv_1_1 = Convolution2D(nb_filter=NUMFAC, nb_row=1, nb_col=1, activation='linear', trainable=False,  border_mode='valid', name="conv_1_1", weights=weights['conv_1_1'])
```
```           
conv_1_1.trainable = False
factor_x = conv_1_1(input_x)
```

but it makes no sense because i have find the corresponding parameters in tf.trainable_variables().

How to set the parameters to be un-trainable?

**Comment by samre12** [comment]
One possible option can be that - when you using an optimizer to compute and apply the gradients in Tensorflow, you could explicitly pass the list of variables for which you wish to compute and apply  the gradients. If no such list is provided, it assumes its value to `tf.trainable_variables()`.
For example, if you are using *Adam Optimizer*, you could use `var_list` in `tf.train.AdamOptimizer().minimize()` to supply the list of trainable variables.

**Comment by HC-2016** [comment]
@samre12 It works. Thank you very much.
